### ISSUE TITLE
Incorrect "Duplicate identifier" error in JS file when JSDoc @typedef followed by parenthetical expression

### ISSUE THREAD

**Issue by davidpcaldwell** [microsoft/TypeScript#36179]
**TypeScript Version:**  3.7.4

**Search terms**
duplicate identifier typedef jsdoc

**Code**
`duplicate-identifier-expression.js`
```js
//@ts-check
/** @typedef {object} NotADuplicateIdentifier */

(2 * 2)
```

**Expected behavior:**
Compiles without error

**Actual behavior:**
```
$ env PATH="${PATH}:../slime/local/jsh/lib/node/bin" ../slime/local/jsh/lib/node/bin/tsc duplicate-identifier-expression.js --allowJs --noemit
duplicate-identifier-expression.js:2:23 - error TS2300: Duplicate identifier 'NotADuplicateIdentifier'.

2 /** @typedef {object} NotADuplicateIdentifier */
                        ~~~~~~~~~~~~~~~~~~~~~~~
Found 1 error.
```

Notes:
* Did not see equivalent error for `.ts` file.
* Did not see equivalent error when expression not enclosed in parentheses.

**Playground Link:**
Could not recreate there.

**Related Issues:**
None found.

**Comment by ajafff** [comment]
The issue is that the same JSDoc is parsed (and bound) for the `ExpressionStatement` and the `ParenthesizedExpression`.

There's another bug where the `ParenthesizedExpression` contains the JSDoc twice: https://ts-ast-viewer.com/#code/PQKhAIAEBcE8AcCmATRAzcBvA9gIwFaIDG0AvuAGLbbgjABQAFAIwCU4A1OMwNxA (check "Show Internals" option and select `ParenthesizedExpression`)

```ts
/** @typedef {object} Foo */
(1) + 1;
```

I'll try to fix the original issue by not parsing JSDoc for ExpressionStatement if it starts with a ParenthesizedExpression. Maybe I'll fix the other parsing issue right away or open a follow-up issue.

**Comment by ajafff** [comment]
It looks like parsing JSDoc twice only happens for JSDoc comments starting in the first line of the file. So it could simply be a duplicate of #29625

**Comment by ajafff** [comment]
Fix is up at #36289
